src/Resources/config/admin.xml

```diff
@@ -14,5 +14,18 @@
             <tag name="form.type" alias="sonata_security_roles"/>
             <argument type="service" id="sonata.user.editable_role_builder"/>
         </service>
+        <service id="sonata.user.roles_matrix_builder" class="Sonata\UserBundle\Security\RolesMatrixBuilder">
+            <argument type="service" id="security.token_storage"/>
+            <argument type="service" id="security.authorization_checker"/>
+            <argument type="service" id="sonata.admin.pool"/>
+            <argument>%security.role_hierarchy.roles%</argument>
+            <call method="setTranslator">
+                <argument type="service" id="translator"/>
+            </call>
+        </service>
+        <service id="sonata.user.form.type.security_roles_matrix" class="Sonata\UserBundle\Form\Type\SecurityRolesMatrixType">
```

Comment:
please rename and use class as id

Reply:
@OskarStark why would we use the class as id?